js/src/Zone.js

```diff
@@ -0,0 +1,57 @@
+import ExpectedMark from "./ExpectedMark";
+import UXCapture from "./UXCapture";
+
+// array holding current, global set of expected zones
+let expectedZones = [];
+
+/**
+ * Zone represents collection of elements groupped together and corresponding phase of page load
+ *
+ * Only one set of zones can be tracked for the same view at one point in time
+ */
+export default class Zone {
```

Comment:
Zones seem to overlap quite a bit with the performance timing concept of a `measure` - do you think it might reduce confusion to call it `Measure`?

Reply:
Not exactly. In fact, I should probably reveal next thing about Zones - they are not really a representation of UserTiming events but rather a collection of UI elements/components/features (all names are ambiguous, unfortunately).

Each View should be loaded in phases that progressively enhance experience revealing one zone after another until full functionality is loaded.

Each phase is measured by capturing events which can be determined using some method. There are two methods I currently know of:
1. collection of Performance Timing APIs events, mostly UserTiming API marks included in the application code, but can also include events available in the browser, like "first paint", for example, or ElementTiming API. This method can be used with synthetic and RUM capture technologies as both can execute JavaScript.
2. video analysis which can determine when each element of the zone was actually painted on the screen and therefore when zone was complete. Detection of boundaries of each component can be accomplished by capturing coordinates of each HTML element zone consists of and using WPT visual progression / speed index algorithms on a modified video frames that only include pixels within those boundaries. This method can only be used with synthetic capture technologies as they require video frame recording.

Both methods have their unique advantages, here are the main ones:
1. UserTiming API can be ran in RUM and therefore is the only way to correlate results with actual business KPIs for specific users
2. Video analysis actually represents what is displayed to the user and exactly aligns with visual user experience requirements while UserTiming is an approximation of browser behavior.

Both methods are necessary to ensure successful implementation in the long run as synthetic capture can calibrate RUM results providing realistic accuracy measures.

Long story short, Zone does not really equate to measure in this library because we'll need to define both UserTiming marks and CSS selectors for zones to accomplish both.

I realized that CSS selectors (or other methods of identifying specific HTML Elements) are not really optional as we were discussing this instrumentation with @sadafie and I was thinking of how we'd work with product managers and designers to begin with.

Reply:
ah, nice. So essentially views are comprise of zones, each zone has a set of measures it's interested in, and each measure is defined by a set of (possibly overlapping) marks.

View : Zone - one to many
Zone : Measure - one to many
Measure : Mark - many to many

I don't want to create too many pieces that we need to track, but I think having those 4 would be valuable, and we can diagram how they interact relatively easily